### PR TITLE
Adding Objective-C support for bazel

### PR DESCRIPTION
**Pull request by romoore**
In order to allow Objective-C bazel libraries to depend on ios-snapshot-test-case, we need a minimum amount of support added to the library.  This includes:

* Defining the BUILD targets for bazel
* Rewriting framework-style imports during build so that the symbols can be resolved.

This change excludes Swift support, since at present we have no code depending on Swift for snapshot tests so it would be difficult to detect breaking changes later.

Supporting work for https://github.com/material-components/material-components-ios/issues/6287

### CODE REVIEW

File: apple_framework_relative_headers.bzl [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Forked from
```
Comment (codeman7) [inline apple_framework_relative_headers.bzl]: Because it's forked does that mean we don't need a copyright?

File: ios_snapshot_test_case.bzl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2019-present The Material Foundation Authors. All Rights Reserved.
```
Comment (codeman7) [inline ios_snapshot_test_case.bzl]: Is this file needed?

**Review by codeman7 — APPROVED** [review]

File: apple_framework_relative_headers.bzl (hunk 1 above)
Comment (romoore) [reply apple_framework_relative_headers.bzl]: I'm not sure - the original had no copyright. I was going to wait until I could ask someone with knowledge before merging.

File: ios_snapshot_test_case.bzl (hunk 2 above)
Comment (romoore) [reply ios_snapshot_test_case.bzl]: It should be removed.

Comment (romoore) [reply ios_snapshot_test_case.bzl]: Done.

File: apple_framework_relative_headers.bzl (hunk 1 above)
Comment (romoore) [reply apple_framework_relative_headers.bzl]: I don't see the same method definition elsewhere and it seems substantial enough that I think it warrants a header. Since both this repo and material-internationalization-ios are under the material-foundation organization, the copyright header would be the same in any case.  I'm going to add it for now (to move forward) and follow-up with @ianegordon to confirm that it makes sense to him as well.
